File: python/square-root/square_root.py
def square_root(number):
    """
    Calculate the square root of a natural radicand using the Newton-Raphson method.

    Args:
        number (int): The number for which the square root is to be determined.

    Returns:
        float: The square root approximation of the radicand.
    """
    for i in range(number + 1):
        if i * i == number:
            return i

File: javascript/square-root/square-root.js
export const squareRoot = (radicand) => {
  if (radicand < 0) {
    return NaN; // Square root of negative numbers is undefined
  }

  if (radicand === 0 || radicand === 1) {
    return radicand;
  }

  let sqrt = 1;
  while (sqrt * sqrt !== radicand) {
    sqrt++;
  }
  return sqrt;
};
